qiskit/synthesis/discrete_basis/commutator_decompose.py

```diff
@@ -53,17 +53,15 @@ def _compute_rotation_axis(matrix: np.ndarray) -> np.ndarray:
     """
     _check_is_so3(matrix)
 
-    trace = _compute_trace_so3(matrix)
-    theta = math.acos(0.5 * (trace - 1))
-    if math.sin(theta) > 1e-10:
-        x = 1 / (2 * math.sin(theta)) * (matrix[2][1] - matrix[1][2])
-        y = 1 / (2 * math.sin(theta)) * (matrix[0][2] - matrix[2][0])
-        z = 1 / (2 * math.sin(theta)) * (matrix[1][0] - matrix[0][1])
-    else:
-        x = 1.0
-        y = 0.0
-        z = 0.0
-    return np.array([x, y, z])
+    # A simple (but probably not the most efficient) way to compute the rotation
```

Comment:
Regarding our offline convo; if you agree then using the Rodrigues path is probably faster 🙂

Reply:
I agree. In [a8a1954](https://github.com/Qiskit/qiskit/pull/14217/commits/a8a1954c097546f24f80ba106c87c7f57e9ac533) I am restoring the previous code while also handling the 180-degree rotation based on your compilation.